src/api/auks/auks_api.c

```diff
@@ -687,13 +689,70 @@ auks_api_get_auks_cred(auks_engine_t * engine,uid_t uid,auks_cred_t* cred)
 	return fstatus;
 }
 
+int auks_api_run_helper(char *helper_script, char *auks_credcache, uid_t uid, gid_t gid)
+{
+	int helper_pid;
+
+	if (helper_script == NULL) {
+		auks_log3("No helper script provided");
+		return AUKS_SUCCESS; /* More of a warning than an error */
+	}
+
+	/* Launch helper renewer script. */
+
+	helper_pid = fork();
+	if (helper_pid == -1) {
+		auks_log2("unable to launch helper renewer script");
+		return AUKS_ERROR;
+	}
+	else if (helper_pid == 0) {
+		/* If we just fork() and exec(), the script will end up with uid=0 and gid=0.
+		 * Therefore, restore uid=0 privileges temporarily so that we can call setgid
+		 * and setuid for a full privilege drop. */
+		seteuid(getuid());
+		setegid(getgid());
+
+		if (setgid(gid)) {
```

Comment:
I think  there's a missing < 0 from the if

Reply:
Only a 0 is a success code, according to the manpage (values > 0 are undefined). Therefore this if is equivalent to `!= 0` rather than `< 0`, although I'm sure that both would work.